docs/proposals/docs-process.md

```diff
@@ -0,0 +1,94 @@
+
+This highly lubricates the ability for external users to make changes, as they can immediately see the consequences of their actions in their forked environment. 
+
+## Why migrate to kubernetes.github.io instead of using a `gh-pages` branch?
+
+GitHub Pages also suppots using branches of an existing repo to host Jekyll sites; in this case you would reuse the repo, but check in the Jekyll site to a branch named `gh-pages`. While you still get auto-building and staging, it will build at `http://username.github.io/**repo**` -- and in our case that is problematic because absolute links to any URI within the site will not work the same as they will on kubernetes.io. If we use the `username.github.io` repo instead of the gh-pages branch, then including an image from `/images/example.png` will work the same on `http://kubernetes.github.io`, `http://your_username.github.io`, `http://kubernetes.io`, and your local machine (running `jekyll serve` off of a clone of our repo).
+
+## Overview of docs migration and how release process will change
+
+Right now, the subdirectory of kubernetes/docs acts as the HEAD for the docs, which is used to populate new releases, as well as the basis for cherrypicking changes into docs for old releases. However, consider a "roaming HEAD" situation, such that the new release is always treated as the basis for populating new releases and  cherrypicks. In otherwords, whatever information is `http://kubernetes.io/vNEWEST/` is HEAD, and we no longer have a dependency on an underlying “master” docset. 
+
+New releases can be developed under `http://kubernetets.io/vNEXT`. We can do this because there is an extensive set of steps to get any one particular folder full of docs to be the default. Or, at worst, we can make a vNEXT branch of the `kubernetes/kubernetes.github.io` repo and do vNEXT work there, culminating in a pull request at release time. But this would only be necessary if we don’t want users to ever see what vNEXT docs look like. I personally vote for doing our work out in public. We can always mark those docs "beta" -- and keeping them in the master branch means that we get to use all the great github.io staging tools. 
+
+## Replacing of munging scripts
+
+- Inline/top-of-page Table of Contents can be done with JavaScript, e.g.: http://jsfiddle.net/s8m2t/8/
+- Header/footer insertion can be done by simply using the native layouts functionality in Jekyll
+- Linking to specific versions of docs can be done with inline variables in the URL, e.g. /docs/{{ version }}/hello-world.md
+- Replacement of file extensions in link paths should not be necessary as public Jekyll URLs are "friendly" (extension-free)
+- Other pre/post-processing scripts for docs?
```

Comment:
We have unit tests (examples_test.go) and e2e tests of examples and user-guide samples. 

One possible approach proposed by @johndmulhausen would be to leave the yaml and other files in kubernetes/master, where we can test them as we do now, and use a jekyll plugin to include them into the markdown for the other repo.

The obvious disadvantage of that approach is that the example doc text and other assets would be spread across multiple repos.

Eventually we need to solve the multiple-repo problem, such as for contrib, the UI, etc. Once we make the CI, e2e, vendoring, etc. work for other repos, maybe we could move all the assets to the doc repo. We'd really like to move the examples to another repo.